src/main/javascript/service/cat-validation-service.js

```diff
@@ -15,54 +17,122 @@ angular.module('cat.service.validation', ['cat.service.message'])
+            catValidationContexts[uuid] = {
+                uuid: uuid,
+                global: undefined,
+                fieldErrors: {},
+                knownFields: []
+            };
+            return uuid;
+        };
+
+        this.destroyContext = function (context) {
+            delete catValidationContexts[context];
+        };
+
+        this.updateFromRejection = function (rejection) {
+            var context;
+            if (!!rejection.config) {
+                context = rejection.config.catValidationContext;
+            }
+
+            var validations = this.getValidations(context);
 
-            var fieldErrors = catValidations.fieldErrors = {};
+            var fieldErrors = validations.fieldErrors = {};
 
             if (!!rejection.data.fieldErrors) {
                 // group by field
                 _.forEach(rejection.data.fieldErrors, function (fieldError) {
-                    fieldErrors[fieldError.field] = fieldErrors[fieldError.field] || [];
-                    fieldErrors[fieldError.field].push(fieldError.message);
+                    // Allow config to switch between displaying errors at the field and displaying errors at known fields or globally
+                    if (catMessagesConfig.knownFieldsActive === true) {
+                        // If the error is for a known field, show the error at the field.
+                        // If not, display it as a global error.
+                        if (_.indexOf(validations.knownFields, fieldError.field) > -1) {
+                            fieldErrors[fieldError.field] = fieldErrors[fieldError.field] || [];
+                            fieldErrors[fieldError.field].push(fieldError.message);
+                        } else {
+                            rejection.data.globalErrors.push(fieldError.message);
+                        }
+                    } else {
+                        fieldErrors[fieldError.field] = fieldErrors[fieldError.field] || [];
+                        fieldErrors[fieldError.field].push(fieldError.message);
+                    }
                 });
             }
 
+            if (!!rejection.data.globalErrors) {
+                validations.global = rejection.data.globalErrors;
+
+                // TODO is this also context dependend? or even necessary?
+                $globalMessages.addMessages('error', rejection.data.globalErrors, context);
+            }
+        };
+
+        this.clearValidationErrors = function (context) {
+            delete this.getValidations(context).global;
+            this.getValidations(context).fieldErrors = {};
         };
 
-        this.clearValidationErrors = function () {
-            delete catValidations.global;
-            catValidations.fieldErrors = {};
+        this.hasGlobalErrors = function (context) {
+            var globalErrors = this.getValidations(context).global;
+            return !!globalErrors && globalErrors.length > 0;
```

Comment:
we check with `!!globalErrors` if the `globalErrors` exists (is not `undefined`). so this does not lead to an error.